sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2567,8 +2567,12 @@ class Dataset[T] private[sql](
   }
 
   private[sql] def collectToPython(): Int = {
+    EvaluatePython.registerPicklers()
     withNewExecutionId {
-      PythonRDD.collectAndServe(javaToPython.rdd)
```

Comment:
Note that `collectAndServe` was internally calling `rdd.collect().iterator`, so this patch's change doesn't have any impact on the maximum number of rows that need to be buffered in the JVM; this would not have been the case if the old code used `collectAndServeIterator`.